app/Models/UserLesson.php

```diff
@@ -13,6 +15,28 @@ class UserLesson extends Model
 
     protected $fillable = [
         'user_id',
-        'lesson_id'
+        'lesson_id',
+        'progress'
     ];
+
+    public static function getUserLesson($userId, $lessonId)
+    {
+        return UserLesson::where('user_id', $userId)->where('lesson_id', $lessonId)->first();
+    }
+
+    public function caculateProgress()
+    {
+        $documentsGet = Auth::user()->documents;
+        $documents = Lesson::find($this['lesson_id'])->documents;
+        $count = 0;
+        foreach ($documents as $document) {
```

Comment:
$documentsGet = Auth::user()->documents->pluck('id')
$documents = Lesson::find($this['lesson_id'])->documents;->pluck('id')